docs/architecture/feature.md

```diff
@@ -3,6 +3,25 @@
 Features consist of two parts; abstraction (port) and implementation (adapter).
 Each adapter exposes a class that implements `IFeature` interface.
 
+`IFeature` has an `Id` property, which is the name of the implementing class
+by default. You can give your feature a custom `Id` by overriding
+`IFeature.Id`.
+
+```csharp
+public class CustomFeature : IFeature
+{
+...
+public string Id => "CustomId".   
+...
+}
+```
+
+`Id` determines uniqueness of features, thus different features should have
+unique ids. When adding features to your application, if two features have the
+same `Id`, you will get an error, since adding the same feature multiple
+times is not allowed. You can refer to conventions when giving custom `Id` to
+your features.
```

Comment:
these rules are explained under specs they are being tested, this is the introduction part